packages/edit-navigation/src/components/block-placeholder/index.js

```diff
@@ -0,0 +1,191 @@
+/**
+ * WordPress dependencies
+ */
+/**
+ * WordPress dependencies
+ */
+import { createBlock } from '@wordpress/blocks';
+import {
+	Placeholder,
+	Button,
+	DropdownMenu,
+	MenuGroup,
+	MenuItem,
+	Spinner,
+} from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import {
+	forwardRef,
+	useCallback,
+	useState,
+	useEffect,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { chevronDown } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { store as navigationStore } from '../../store';
+import { useMenuEntityProp } from '../../hooks';
+import useNavigationEntities from './use-navigation-entities';
+import menuItemsToBlocks from './menu-items-to-blocks';
+
+/**
+ * Convert pages to blocks.
+ *
+ * @param {Object[]} pages An array of pages.
+ *
+ * @return {WPBlock[]} An array of blocks.
+ */
+function convertPagesToBlocks( pages ) {
+	if ( ! pages?.length ) {
+		return null;
+	}
+
+	return pages.map( ( { title, type, link: url, id } ) =>
+		createBlock( 'core/navigation-link', {
+			type,
+			id,
+			url,
+			label: ! title.rendered ? __( '(no title)' ) : title.rendered,
+			opensInNewTab: false,
+		} )
+	);
+}
+
+function BlockPlaceholder( { onCreate }, ref ) {
+	const [ selectedMenu, setSelectedMenu ] = useState();
+	const [ isCreatingFromMenu, setIsCreatingFromMenu ] = useState( false );
+
+	const selectedMenuId = useSelect( ( select ) =>
+		select( navigationStore ).getSelectedMenuId()
+	);
```

Comment:
```suggestion
	const [ selectedMenuId ] = useSelectedMenuId();
```

We could use our custom hook here.